sherpa/plot/backend_utils.py

```diff
@@ -0,0 +1,200 @@
+#
+#  Copyright (C) 2022
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+'''This module has utility functions for backends to format doctrings or validate inputs.'''
+
+import functools
+from inspect import signature
+import re
+
+__all__ = ('translate_args', 'add_kwargs_to_doc', 'get_keyword_defaults')
+
+kwargs_indent = re.compile("\n(?P<spaces>\s*){kwargs}",flags=re.M)
```

Comment:
I'm getting this warning from `pytest`:

```
sherpa/plot/backend_utils.py:28
  /home/dburke/sherpa/sherpa-master/sherpa/plot/backend_utils.py:28: DeprecationWarning: invalid escape sequence \s
    kwargs_indent = re.compile("\n(?P<spaces>\s*){kwargs}",flags=re.M)
```

Reply:
I think that's pytest specific. At least in the version of Python I use, the `re` module will take the string with or without raw escaping and find identical matches.
```
In [1]: t='''
   ...: asdf
   ...: asdf
   ...: 
   ...:     {kwargs}
   ...:     '''

In [2]: import re

In [3]: kwargs_indent = re.compile("\n(?P<spaces>\s*){kwargs}",flags=re.M)

In [4]: kwargs_indent.findall(t)
Out[4]: ['\n    ']

In [5]: kwargs_indent = re.compile(r"\n(?P<spaces>\s*){kwargs}",flags=re.M)

In [6]: kwargs_indent.findall(t)
Out[6]: ['\n    ']
```
I can add the raw escaping which will hopefully make that error go away for you.

Reply:
I thought this was a Python warning: from https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals

```
Changed in version 3.6: Unrecognized escape sequences produce a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning). In a future Python version they will be a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning) and eventually a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError).
```

it's just that `pytest` reports this warning but `python setup.py test` and `sherpa_test` don't seem to.